evalml/tests/model_understanding_tests/test_partial_dependence.py

```diff
@@ -3008,3 +3008,123 @@ def test_partial_dependence_dfs_transformer_does_not_calculate_feature_matrix(
     pd.testing.assert_frame_equal(part_dep, fast_part_dep)
+
+
+@pytest.mark.parametrize(
+    "nullable_y_ltype",
+    ["IntegerNullable", "AgeNullable", "BooleanNullable"],
+)
+@pytest.mark.parametrize(
+    "grid_resolution",
+    [2, 10, 50],
+)
+@pytest.mark.parametrize(
+    "use_int_null_many_values",
+    [True, False],
+)
+def test_partial_dependence_with_nullable_types(
+    nullable_type_test_data,
+    nullable_type_target,
+    linear_regression_pipeline,
+    logistic_regression_binary_pipeline,
+    nullable_y_ltype,
+    grid_resolution,
+    use_int_null_many_values,
+):
+    y = nullable_type_target(ltype=nullable_y_ltype, has_nans=False)
+    X = nullable_type_test_data(has_nans=False)
+    X = X.ww.select(["numeric", "Boolean", "BooleanNullable"])
+
+    int_col = "int col nullable"
+    if use_int_null_many_values:
+        X.ww[int_col] = ww.init_series(
+            pd.Series(range(len(y))),
+            logical_type="IntegerNullable",
+        )
+
+    pipeline = linear_regression_pipeline
+    if nullable_y_ltype == "BooleanNullable":
+        pipeline = logistic_regression_binary_pipeline
+
+    pipeline.fit(X, y)
+    part_dep = partial_dependence(
+        pipeline,
+        X,
+        grid_resolution=grid_resolution,
+        features=int_col,
+    )
+    assert len(part_dep["partial_dependence"]) == min(
+        grid_resolution,
+        len(X[int_col].dropna().unique()),
+    )
+    assert len(part_dep["feature_values"]) == min(
+        grid_resolution,
+        len(X[int_col].dropna().unique()),
+    )
+    assert not part_dep.isnull().any(axis=None)
+
+    fast_part_dep = partial_dependence(
+        pipeline,
+        X,
+        features=int_col,
+        grid_resolution=grid_resolution,
+        fast_mode=True,
+        X_train=X,
+        y_train=y,
+    )
+    pd.testing.assert_frame_equal(part_dep, fast_part_dep)
+
+    part_dep = partial_dependence(
+        pipeline,
+        X,
+        grid_resolution=grid_resolution,
+        features="bool col nullable",
+    )
+    assert len(part_dep["partial_dependence"]) == 2
+    assert len(part_dep["feature_values"]) == 2
+    assert not part_dep.isnull().any(axis=None)
+
+    fast_part_dep = partial_dependence(
+        pipeline,
+        X,
+        features="bool col nullable",
+        grid_resolution=grid_resolution,
+        fast_mode=True,
+        X_train=X,
+        y_train=y,
+    )
+    pd.testing.assert_frame_equal(part_dep, fast_part_dep)
+
+
+@pytest.mark.parametrize(
+    "numeric_ltype",
+    ["IntegerNullable", "Integer", "Age", "AgeNullable", "AgeFractional", "Double"],
+)
+@pytest.mark.parametrize(
+    "grid_resolution",
+    [2, 10, 50],
+)
+def test_partial_dependence_grid_values_for_numeric_data(
+    logistic_regression_binary_pipeline,
+    numeric_ltype,
+    grid_resolution,
+):
+    y = ww.init_series(pd.Series([True, False] * 25), logical_type="Boolean")
+    X = pd.DataFrame({"col": pd.Series(range(len(y)))})
```

Comment:
Does this test fail on main? It looks like with a dataset length of 50, none of the parameterized grid resolution values would result in fractions.

Reply:
The age fraction and double tests pass on main, but all the remaining tests currently fail. Age and Integer fail on the check for fractional values for all of the grid resolution parameters (more on that in a sec), and the AgeNullable and IntegerNullable types fail because of the type conversion error.

Regarding your point about the grid_resolution parameterization as it relates to the length/number of unique values in X being 50 - you're right that that's not what I want, but they're all currently resulting in fractions. When the grid resolution is strictly lower than the number of unique values, we calculate our own grid values, which is where the fractional values come from.

I'm dealing with an off by one error here - I want the final number to be greater than 50. And in that case, I would expect those tests to pass on main, which I'm fine with. This test is both about recording the existing behavior and fixing broken behavior.

I'm going to change the final grid resolution number to be 60.